f5/bigip/tm/ltm/policy.py

```diff
@@ -56,6 +59,24 @@ def __init__(self, policy_s):
         self._meta_data['required_creation_parameters'].update(('strategy',))
         temp = {'tm:ltm:policy:rules:rulescollectionstate': Rules_s}
         self._meta_data['attribute_registry'] = temp
+        self._meta_data['optional_parameters'] = {'rules': ['description']}
+
+    def _filter_version_specific_options(self, tmos_ver, **kwargs):
+        '''Filter version-specific optional parameters
+
+        Some optional parameters only exist in v12.1.0 and greater,
+        filter these out for earlier versions to allow backward comatibility.
+        '''
+
+        if LooseVersion(tmos_ver) < LooseVersion('12.1.0'):
+            for k, parms in self._meta_data['optional_parameters'].items():
+                for r in kwargs.get(k, []):
+                    for parm in parms:
+                        value = r.pop(parm, None)
+                        if value is not None:
+                            logger.info(
```

Comment:
I would suggest raising an error, it is easier to handle by the caller

Reply:
I discussed this with @pjbreaux and @zancas and we agreed that the caller should not receive an exception. SDK should handle the BIG-IP version compatibility issue by stripping the optional parameter and logging the event.

Reply:
This is taken from our dev playbook, which is what we are adhering to, so what has changed? specifying a non existent parameter is a failure

````
Failures generate exceptions
    If the SDK enters a surprising or unexpected state it raises an exception. That's it. It's not generally up to the SDK to implement decision logic that handles edge-cases.. EXCEPT where the SDK is smoothing known issues in the device REST server. (See below.)

````